src/core/lib/iomgr/call_combiner.cc

```diff
@@ -39,10 +39,42 @@ static gpr_atm encode_cancel_state_error(grpc_error* error) {
   return static_cast<gpr_atm>(1) | (gpr_atm)error;
 }
 
+#ifdef GRPC_TSAN_ENABLED
+static void tsan_closure(void* user_data, grpc_error* error) {
+  grpc_call_combiner* call_combiner =
+      static_cast<grpc_call_combiner*>(user_data);
+  grpc_core::RefCountedPtr<grpc_call_combiner::TsanLock> lock;
+  if (!call_combiner->tsan_lock->taken.load(std::memory_order_relaxed)) {
+    lock = call_combiner->tsan_lock;
+    TSAN_ANNOTATE_RWLOCK_ACQUIRED(lock.get(), true);
+    lock->taken.store(true, std::memory_order_relaxed);
+  }
+  GRPC_CLOSURE_RUN(call_combiner->original_closure, GRPC_ERROR_REF(error));
+  if (lock != nullptr) {
+    lock->taken.store(false, std::memory_order_relaxed);
+    TSAN_ANNOTATE_RWLOCK_RELEASED(lock.get(), true);
+  }
+}
+#endif
+
+static void combiner_sched_closure(grpc_call_combiner* combiner,
```

Comment:
Please change "combiner" to "call_combiner" in both the function name and the name of the first parameter.  (Confusingly, we have a separate data structure called combiner, which is actually unrelated to call_combiner.)